vital/types/polygon.h

```diff
@@ -0,0 +1,214 @@
+/*ckwg +29
+ * Copyright 2016 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief Abstract polygon interface
+ */
+
+#ifndef VITAL_TYPES_POLYGON_H
+#define VITAL_TYPES_POLYGON_H
+
+#include <vital/vital_config.h>
+#include <vital/vital_export.h>
+#include <vital/types/vector.h>
+
+#include <memory>
+
+namespace kwiver {
+namespace vital {
+
+class vital_polygon;
+typedef std::shared_ptr<vital_polygon> vital_polygon_sptr;
+
+// ----------------------------------------------------------------
+/**
+ * @brief Abstract base polygon class.
+ *
+ * This class represents a polygon with a limited number of
+ * attributes. The concrete implementation of the polygon is delegated
+ * to a concrete derived class. There may be more than one possible
+ * implementation. These implementations should provide a way to
+ * access the implementation specific methods because they usually
+ * provide additional attributes and operations. These derived classes
+ * should supply conversion methods to and from the basic (core)
+ * implementation.
+ *
+ * This class behaviour is considered the specification for all
+ * derived classes.
+ */
+
+class VITAL_EXPORT polygon
```

Comment:
As we discussed, this class is probably simple and small enough that we don't need the abstract base class treatment. We do, however, want to keep conversion functions to and from vgl_polygon in the vxl arrow.  If we later have an algorithm API need for passing around large collections of polygons it may make sense to create a "polygon_set" class that using an abstract base class to avoid copying all the data.